common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -190,67 +228,61 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (appType != null) {
             builder.field(APP_TYPE_FIELD, appType);
         }
+        // is_hidden field isn't going to be set by user. It will be set by the code.
+        if (isHidden != null) {
+            builder.field(MLModel.IS_HIDDEN_FIELD, isHidden);
+        }
         builder.endObject();
         return builder;
     }
 
     public static MLAgent parse(XContentParser parser) throws IOException {
-        String name = null;
-        String type = null;
-        String description = null;;
+        return parseCommonFields(parser, true); // true to parse isHidden field
+    }
+
+    public static MLAgent parseFromUserInput(XContentParser parser) throws IOException {
+        return parseCommonFields(parser, false); // false to skip isHidden field
+    }
+
+    private static MLAgent parseCommonFields(XContentParser parser, boolean parseHidden) throws IOException {
+        String name = null, type = null, description = null;
         LLMSpec llm = null;
         List<MLToolSpec> tools = null;
         Map<String, String> parameters = null;
         MLMemorySpec memory = null;
-        Instant createdTime = null;
-        Instant lastUpdateTime = null;
+        Instant createdTime = null, lastUpdateTime = null;
```

Comment:
Why change the code style ? Using consistent code style can make it easier to read.